File: ai-hub/frontend/src/types/conversation.ts
/** 会话类型定义 */
export interface Conversation {
  id: string
  title: string
  createdAt: string
  updatedAt: string
}

export interface MessageData {
  id: string
  conversationId: string
  role: string
  content: string
  metadata: string
  createdAt: string
}

File: ai-hub/backend/app/common/api/schemas/conversation.py
"""会话相关 Schema"""

from typing import Any

from pydantic import BaseModel


class ConversationCreate(BaseModel):
  """创建会话请求"""
  title: str = "新会话"
  type: str = "chat"
  metadata: dict[str, Any] | None = None


class ConversationUpdate(BaseModel):
  """更新会话请求"""
  title: str


class ConversationResponse(BaseModel):
  """会话响应"""
  id: str
  user_id: str = ""
  title: str
  type: str = "chat"
  metadata: str = "{}"
  created_at: str
  updated_at: str


class MessageResponse(BaseModel):
  """消息响应"""
  id: str
  conversation_id: str
  role: str
  content: str
  metadata: str
  created_at: str
